x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/actions_log_filter.tsx

```diff
@@ -25,19 +25,79 @@ export const ActionsLogFilter = memo(
     onChangeFilterOptions: (selectedOptions: string[]) => void;
   }) => {
     const getTestId = useTestIdGenerator('response-actions-list');
+
+    // popover states and handlers
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+    const onPopoverButtonClick = useCallback(() => {
+      setIsPopoverOpen(!isPopoverOpen);
+    }, [setIsPopoverOpen, isPopoverOpen]);
+    const onClosePopover = useCallback(() => {
+      setIsPopoverOpen(false);
+    }, [setIsPopoverOpen]);
+
+    // search string state
+    const [searchString, setSearchString] = useState('');
     const {
+      areHostsSelectedOnMount,
+      isLoading,
       items,
       setItems,
       hasActiveFilters,
       numActiveFilters,
       numFilters,
+      setAreHostsSelectedOnMount,
       setUrlActionsFilters,
+      setUrlHostsFilters,
       setUrlStatusesFilters,
-    } = useActionsLogFilter(filterName, isFlyout);
+    } = useActionsLogFilter({
+      filterName,
+      isFlyout,
+      isPopoverOpen,
+      searchString,
+    });
+
+    // track popover state to pin selected options
+    const wasPopoverOpen = useRef(isPopoverOpen);
+    useEffect(() => {
+      return () => {
+        wasPopoverOpen.current = isPopoverOpen;
+      };
+    }, [isPopoverOpen, wasPopoverOpen]);
+
+    // compute if selected hosts should be pinned
+    const shouldPinSelectedHosts = useCallback(
+      (isNotChangingOptions: boolean = true) => {
+        // case 1: when no hosts are selected initially
+        return (
+          isNotChangingOptions && wasPopoverOpen.current && isPopoverOpen && filterName === 'hosts'
+        );
+      },
+      [filterName, isPopoverOpen, wasPopoverOpen]
```

Comment:
`wasPopoverOpen` could be removed from the dependency arrays, as it has no effect on re-render.

Reply:
Good catch. Will update it.

Reply:
e2cb1aa2caa06ae78f434e2e16889046fb015a5a